lib/BaseTemplateController.js

```diff
@@ -32,8 +32,9 @@ module.exports = class BaseTemplateController extends CompositeController {
     let key = '';
     if (valueFrom['secretKeyRef']) {
       let secretName = objectPath.get(valueFrom, 'secretKeyRef.name');
+      let namespace = objectPath.get(valueFrom, 'secretKeyRef.namespace');
```

Comment:
Suggest ` let namespace = objectPath.get(valueFrom, 'secretKeyRef.namespace');` changed to 
` let namespace = objectPath.get(valueFrom, 'secretKeyRef.namespace',this.namespace);`
then any future reference to namespace always has a valid value...

Reply:
duh! dont know why i didnt see that to begin with